lua/nvim-treesitter/utils.lua

```diff
@@ -93,4 +93,63 @@ function M.smallest_containing_scope(node, bufnr)
   return current or root
 end
 
+--- Get next node with same parent
+-- @param node                 node
+-- @param allow_switch_parents allow switching parents if last node
+-- @param allow_next_parent    allow next parent if last node and next parent without children
+function M.get_next_node(node, allow_switch_parents, allow_next_parent)
+ local destination_node
+  local parent = node:parent()
+
+  if parent then
+    local found_pos = 0
+    for i = 0,parent:named_child_count()-1,1 do
+      if parent:named_child(i) == node then
+        found_pos = i
+        break
+      end
+    end
+    if parent:named_child_count() > found_pos + 1 then
+      destination_node = parent:named_child(found_pos + 1)
+    elseif allow_switch_parents then
+      local next_node = M.get_next_node(node:parent())
+      if next_node and next_node:named_child_count() > 0 then
+        destination_node = next_node:named_child(0)
+      elseif next_node and allow_next_parent then
+        destination_node = next_node
+      end
+    end
+  end
+  return destination_node
+end
+
+--- Get previous node with same parent
+-- @param node                     node
+-- @param allow_switch_parents     allow switching parents if first node
+-- @param allow_previous_parent    allow previous parent if first node and previous parent without children
+function M.get_previous_node(node, allow_switch_parents, allow_previous_parent)
+  local destination_node
+  local parent = node:parent()
+  if parent then
+    local found_pos = 0
+    for i = 0,parent:named_child_count()-1,1 do
+      if parent:named_child(i) == node then
+        found_pos = i
+        break
+      end
+    end
+    if 0 <= found_pos - 1 then
```

Comment:
`if 0 < found_pos`

Reply:
Why? 0 is a valid index for a node. It's c indices not lua.

Reply:
Sorry I am confused. Now I get what you mean! :smile: